editor/gui/editor_range_dial.cpp

```diff
@@ -0,0 +1,317 @@
+					const Size2 min_value_string_size = font->get_string_size(min_value_str, HORIZONTAL_ALIGNMENT_CENTER, -1.0f, font_size);
+
+					if (number_x_max - max_value_string_size.x / 2 < get_rect().size.x && number_x_max + max_value_string_size.x / 2 > 0) {
+						draw_line(Point2(number_x_max, 0), Point2(number_x_max, get_rect().size.y - max_value_string_size.y), accent_color);
+						draw_string(font, Point2(number_x_max - (max_value_string_size.x / 2), get_rect().size.y).floor(), max_value_str, HORIZONTAL_ALIGNMENT_CENTER, -1, font_size, accent_color);
+					}
+					if (number_x_min + min_value_string_size.x > 0 && number_x_min - min_value_string_size.x < get_rect().size.x) {
+						draw_line(Point2(number_x_min, 0), Point2(number_x_min, get_rect().size.y - max_value_string_size.y), accent_color);
+						draw_string(font, Point2(number_x_min - (min_value_string_size.x / 2), get_rect().size.y).floor(), min_value_str, HORIZONTAL_ALIGNMENT_CENTER, -1, font_size, accent_color);
+					}
+				}
+
+				if (!lines.is_empty()) {
+					mono_color.a = 0.5 * alpha;
+					draw_multiline(lines, mono_color);
+				}
+				if (!lines_outside.is_empty()) {
+					mono_color_outside.a = 0.25 * alpha;
+					draw_multiline(lines_outside, mono_color_outside);
+				}
+				if (!lines_highlight.is_empty()) {
+					draw_multiline(lines_highlight, mono_color_highlight);
+				}
+				if (!lines_highlight_outside.is_empty()) {
+					draw_multiline(lines_highlight_outside, mono_color_highlight_outside);
+				}
+				if (!lines_highlight_outside.is_empty()) {
+					draw_multiline(lines_highlight_outside, mono_color_highlight_outside);
+				}
+			}
+			draw_line(Vector2(get_rect().size.x / 2, 0), Vector2(get_rect().size.x / 2, get_rect().size.y), Color(0.96, 0.2, 0.32));
+		}
+	}
+}
+
+void EditorRangeDial::_bind_methods() {
+	ClassDB::bind_method(D_METHOD("get_zoom"), &EditorRangeDial::get_zoom);
+	ClassDB::bind_method(D_METHOD("set_zoom", "amount"), &EditorRangeDial::set_zoom);
+
+	ClassDB::bind_method(D_METHOD("add_zoom_from_mouse_dist", "relative"), &EditorRangeDial::add_zoom_from_mouse_dist);
+	ClassDB::bind_method(D_METHOD("scale_diff", "diff"), &EditorRangeDial::scale_diff);
+
+	ClassDB::bind_method(D_METHOD("set_highlighting_range", "enable"), &EditorRangeDial::set_highlighting_range);
+	ClassDB::bind_method(D_METHOD("is_highlighting_range"), &EditorRangeDial::is_highlighting_range);
+
+	ADD_SIGNAL(MethodInfo("zoom_changed"));
+	ADD_PROPERTY(PropertyInfo(Variant::FLOAT, "zoom"), "set_zoom", "get_zoom");
+	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "highlighting_range", PROPERTY_HINT_NONE, "", PROPERTY_USAGE_DEFAULT | PROPERTY_USAGE_EDITOR), "set_highlighting_range", "is_highlighting_range");
+}
+
+void EditorRangeDial::set_zoom(double p_zoom) {
+	double min_zoom;
+
+	double zout = Math::abs(get_min()) + get_max();
+	double mx = EDITOR_GET("interface/inspector/max_drag_zoom_out_value");
```

Comment:
This shouldn't be fetched every time I'd say, it should be loaded when settings change